src/content/docs/en/recipes/call-endpoints.mdx

```diff
@@ -0,0 +1,37 @@
+---
+title: Call endpoints from the server
+description: Learn how to call endpoints from the server in Astro.
+i18nReady: true
+type: recipe
+---
+
+## Recipe
+
+1. Create an endpoint (e.g. `src/pages/api/hello.ts`):
+
+    ```ts title="src/pages/api/hello.ts"
+    import type { APIRoute } from 'astro'
+
+    export const get: APIRoute = () => {
+      return {
+        body: JSON.stringify({
+          hello: 'world',
+        }),
+      }
+    }
+    ```
+
+2. Import the method from the endpoint & call it from a page (e.g. `src/pages/index.astro`). Await the result in the case it's a promise, and convert it to a `Response` if it's not already. Then, you can use the `Response` methods to get the data you need:
+
+    ```tsx title="src/pages/index.astro"
+    ---
+    import { get } from './api/hello.ts'
+
+    let res = await get(Astro)
+    if (!(res instanceof Response)) res = new Response(res.body, res)
+
+    const data = await res.json()
+    ---
+
+    <h1>Hello {data.hello}</h1>
+    ```
```

Comment:
If you are using this method to fetch data from an API and are unsure about the type of data you'll receive, you can add a line to convert it to a `Response` if it's not already. 

``````suggestion {5}
```astro title="src/pages/index.astro" 
---
import { get } from './api/hello.ts'

let response = await get(Astro)
if (!(response instanceof Response)) res = new Response(response.body, response)

const data = await response.json()
---

<h1>{data.greeting} world!</h1>
```
``````
You could do something like this to show adding the new line